src/frontend/src/catalog/source_catalog.rs

```diff
@@ -149,3 +149,9 @@ impl OwnedByUserCatalog for SourceCatalog {
         self.owner
     }
 }
+
+impl std::hash::Hash for SourceCatalog {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.id.hash(state);
+    }
+}
```

Comment:
Ditto

Reply:
There is a test failed https://buildkite.com/risingwavelabs/pull-request/builds/50876#018fcfe9-6fd2-47a7-b266-db3f542322e6. @chenzl25  Do you think this change will affect the test case? It works well on non-single-node model.

Reply:
It shouldn't affect the plan. Can you rerun this test case to see whether it is constantly failed?

Reply:
It was run twice. Failed at the same position.

Reply:
If it shouldn't affect, you can modify the run-e2e-test.sh script first. 
If its running under single node mode, skip the failing test.
Then open an issue to track it. I'll have a look when I'm back.

Reply:
Maybe we can compare the plan of `select * from i1 join i2 using(v1);` in the test case between this PR and the main

Reply:
The plan for create table, index and select looks exactly the same. Maybe not the planner issue

Reply:
> Then open an issue to track it. I'll have a look when I'm back.

Thank you @kwannoel